apis/r/tests/testthat/test-SOMADataFrame.R

```diff
@@ -232,3 +232,76 @@ test_that("soma_joinid validations", {
     "soma_joinid field must be of type Arrow int64"
   )
 })
+
+test_that("platform_config is respected", {
+  uri <- withr::local_tempdir("soma-dataframe")
+
+  # Set Arrow schema
+  asch <- arrow::schema(
+    arrow::field("soma_joinid", arrow::int64(), nullable = FALSE),
+    arrow::field("i32", arrow::int32(), nullable = FALSE),
+    arrow::field("f64", arrow::float64(), nullable = FALSE),
+    arrow::field("utf8", arrow::large_utf8(), nullable = FALSE)
+  )
+
+  # Set tiledb create options
+  cfg <- PlatformConfig$new()
+  cfg$set('tiledb', 'create', 'dataframe_dim_zstd_level', 8)
+  cfg$set('tiledb', 'create', 'sparse_nd_array_dim_zstd_level', 9)
+  cfg$set('tiledb', 'create', 'capacity', 8000)
+  cfg$set('tiledb', 'create', 'tile_order', 'COL_MAJOR')
+  cfg$set('tiledb', 'create', 'cell_order', 'UNORDERED')
+  cfg$set('tiledb', 'create', 'offsets_filters', list("RLE"))
+  cfg$set('tiledb', 'create', 'validity_filters', list("RLE", "NONE"))
+  cfg$set('tiledb', 'create', 'dims', list(
+    soma_joinid = list(
+      filters = list("RLE", list(name="ZSTD", COMPRESSION_LEVEL=9), "NONE")
+      # TODO: test setting/checking tile extent, once shapes/domain-maxes are made programmable.
+      # At present we get:
+      #
+      #   Error: Tile extent check failed; domain max expanded to multiple of tile extent exceeds
+      #   max value representable by domain type
+      #
+      # tile = 999
+    )
+  ))
+  cfg$set('tiledb', 'create', 'attrs', list(
+    i32 = list(
+      filters = list("RLE", list(name="ZSTD", COMPRESSION_LEVEL=9))
+    ),
+    f64 = list(
+      filters = list()
+    )
+  ))
+
+  # Create the SOMADataFrame
+  sdf <- SOMADataFrameCreate(uri=uri, schema=asch, index_column_names=c("soma_joinid"), platform_config = cfg)
+
+  # Read back and check the array schema against the tiledb create options
+  arr <- tiledb::tiledb_array(uri)
+  tsch <- tiledb::schema(arr)
+
+  # TODO: zstd levels x 2
```

Comment:
@eddelbuettel -- I want to check the actual compression level of the i'th filter

However, I still haven't found the tiledb-r syntax that lets me do that ...

Reply:
Do we have an example of something that has an i-th filter?  _Ex ante_ a filter list is a list, so the i-th elem should be a filter we can query as if it wasn;t in a list.

Reply:
@eddelbuettel precisely for unit-testing, right here, I want to get the ith filter and check filter name, compression level, etc to verify the schema got what the platform config requested

i'd love to treat these as lists but i cannot find out how to successfully do so:

```
> arr <- tiledb::tiledb_array('./obs')

> tsch <- tiledb::schema(arr)

> o <- tiledb::filter_list(tsch)$offsets

> o
tiledb_filter_list(c(tiledb_filter("DOUBLE_DELTA"), tiledb_filter_set_option(tiledb_filter("BIT_WIDTH_REDUCTION"),"BIT_WIDTH_MAX_WINDOW",256), tiledb_filter_set_option(tiledb_filter("ZSTD"),"COMPRESSION_LEVEL",-1)))

> length(o)
[1] 1

> o[[1]]
Error in o[[1]] : this S4 class is not subsettable

> typeof(o)
[1] "S4"
```

Reply:
(Give me one more sec.  As the pretty printer shows _we can_ get those parts.)

Reply:
It's a somewhat tricky (sorry!) mix of R _and C++_ indexing:

```r
> f2
tiledb_filter_list(c(tiledb_filter("DOUBLE_DELTA"), tiledb_filter_set_option(tiledb_filter("BIT_WIDTH_REDUCTION"),"BIT_WIDTH_MAX_WINDOW",256), tiledb_filter_set_option(tiledb_filter("ZSTD"),"COMPRESSION_LEVEL",-1))) 
> nfilters(f2)
[1] 3
> f2[1]
tiledb_filter_set_option(tiledb_filter("BIT_WIDTH_REDUCTION"),"BIT_WIDTH_MAX_WINDOW",256) 
> f2[0]
tiledb_filter("DOUBLE_DELTA") 
> f2[1]
tiledb_filter_set_option(tiledb_filter("BIT_WIDTH_REDUCTION"),"BIT_WIDTH_MAX_WINDOW",256) 
> tiledb_filter_get_option(f2[1], "BIT_WIDTH_MAX_WINDOW")
[1] 256
> 
```

Reply:
As one example from the top:

```r
> f2
tiledb_filter_list(c(tiledb_filter("DOUBLE_DELTA"), tiledb_filter_set_option(tiledb_filter("BIT_WIDTH_REDUCTION"),"BIT_WIDTH_MAX_WINDOW",256), tiledb_filter_set_option(tiledb_filter("ZSTD"),"COMPRESSION_LEVEL",-1))) 
> nfilters(f2)
[1] 3
> f2[1]
tiledb_filter_set_option(tiledb_filter("BIT_WIDTH_REDUCTION"),"BIT_WIDTH_MAX_WINDOW",256) 
> f2[0]
tiledb_filter("DOUBLE_DELTA") 
> f2[1]
tiledb_filter_set_option(tiledb_filter("BIT_WIDTH_REDUCTION"),"BIT_WIDTH_MAX_WINDOW",256) 
> tiledb_filter_get_option(f2[1], "BIT_WIDTH_MAX_WINDOW")
[1] 256
> 
```

So we index the outer list by name to pick (here) for example the "offsets" filter.  _That gets another list!_. In that list we index (zero-based, just to be more confusing!) and then get eg the bit-with reduction filter whose coef we can pluck.

Ditto, say, for the (unset, default value) compression level:

```r
> tiledb_filter_get_option( flst[["offsets"]][2], "COMPRESSION_LEVEL")
[1] -1
> 
```